forms-flow-web/src/components/Form/Item/Edit.js

```diff
@@ -232,45 +288,48 @@ useEffect(() => {
   };
   const handleConfirmSettings = () => {
     const parentFormId = processListData.parentFormId;
-      const mapper = {
+    const mapper = {
       formId: form._id,
-      formName: form.title,
-      description: formDescription,
+      formName: formDetails?.name,
+      description: formDetails?.description,
       status: processListData.status || "inactive",
       taskVariables: processListData.taskVariables
         ? processListData.taskVariables
         : [],
-        anonymous: formAccess[0]?.roles.includes(roleIds.ANONYMOUS),
-        parentFormId: parentFormId,
-      formType: form.type,
+      anonymous: rolesState.create.isPublic,
+      parentFormId: parentFormId,
+      formType: formType,
       processKey: workflow?.value,
       processName: workflow?.name,
       id: processListData.id,
       workflowChanged: false,
       statusChanged: false,
       resourceId: form._id,
     };
-    
+
     const authorizations = {
       application: {
         resourceId:parentFormId ,
         resourceDetails: {},
-        roles: []
-    },
+        roles: rolesState.view.selectedOption === "specifiedRoles" ? rolesState.view.selectedRoles : [],
+        ...(rolesState.view.selectedOption === "submitter" && { userName: preferred_userName }) // TBD
+      },
       designer: {
         resourceId: parentFormId,
         resourceDetails: {},
-        roles: []
+        roles: rolesState.edit.selectedOption === "specifiedRoles" ? rolesState.edit.selectedRoles : [],
+        ...(rolesState.edit.selectedOption === "onlyYou" && { userName: preferred_userName })
     },
-    form: {
-      resourceId: parentFormId,
-      resourceDetails: {},
-      roles: []
-  }
-};
-    dispatch(saveFormProcessMapperPut({mapper, authorizations}));
+      form: {
+        resourceId: parentFormId,
+        resourceDetails: {},
+        roles: rolesState.create.selectedOption === "specifiedRoles" ? rolesState.create.selectedRoles : [],
+      }
     };
-  
+    dispatch(saveFormProcessMapperPut({ mapper, authorizations }));
```

Comment:
please check we are callingg formio api also along with this . using Promise.all so that both api calls will happens parallely